### PR TITLE
Adding interface for NAG LP and SOCP solvers.

### PR DESCRIPTION
**Pull request by shuanghuabai**
Hi, could you please pull to merge interface and tests for NAG LP and SOCP solvers. Further modification on documentation will follow. Any problems please let me know. Thank a lot!

### CODE REVIEW

**Comment by SteveDiamond** [comment]
This looks quite good! You found all the places in the code you need to update. The only other places you need to update are doc/source/install/index.rst and doc/source/tutorial/advanced/index.rst. Also though you need to run "flake8 ." from the root folder and fix all the warnings.

**Comment by shuanghuabai** [comment]
I have edited the doc and resolved the warnings from flake8. Thanks a lot!

**Comment by SteveDiamond** [comment]
Thanks! You updated all the docs exactly right. Can you agree to the CLA license?

File: cvxpy/reductions/solvers/conic_solvers/nag_conif.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+class NAG(ConicSolver):
+    """ An interface to the NAG SOCP solver
+    """
+
+    MIP_CAPABLE = False
+    SUPPORTED_CONSTRAINTS = ConicSolver.SUPPORTED_CONSTRAINTS + [SOC]
+
+    # Map of NAG status to CVXPY status
+    STATUS_MAP = {0: s.OPTIMAL}
```
Comment (SteveDiamond) [inline cvxpy/reductions/solvers/conic_solvers/nag_conif.py]: Is this really the only possible result? What about infeasibility and unboundedness?

File: cvxpy/reductions/solvers/conic_solvers/nag_conif.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+        if hs:
+            data[s.H] = np.hstack(tuple(hs))
+        else:
+            data[s.H] = np.array([])
+
+        return (data, inv_data)
+
+    def invert(self, solution, inverse_data):
+
+        STATUS_MAP = {0: s.OPTIMAL,
```
Comment (SteveDiamond) [inline cvxpy/reductions/solvers/conic_solvers/nag_conif.py]: Define the STATUS_MAP once, up top, for the class.

Comment (shuanghuabai) [reply cvxpy/reductions/solvers/conic_solvers/nag_conif.py]: No problem at all. Done. Thanks.

**Comment by SteveDiamond** [comment]
Ok, we're good to go! Thanks for your great work.

**Comment by shuanghuabai** [comment]
Thanks a lot!

**Comment by SteveDiamond** [comment]
@shuanghuabai I've been working on updating the NAG interface to work in CVXPY 1.1 (on the 1.1alpha branch). I haven't been able to get a copy of the NAG solver though. Is there anything you can do to help?

**Comment by shuanghuabai** [comment]
> 
> 
> @shuanghuabai I've been working on updating the NAG interface to work in CVXPY 1.1 (on the 1.1alpha branch). I haven't been able to get a copy of the NAG solver though. Is there anything you can do to help?

@SteveDiamond Sorry I was on annual leave last week. I will try to see if we could help and get back to you.

**Pull request merged**